src/biokbase/narrative/tests/test_appmanager.py

```diff
@@ -900,88 +917,77 @@ def test_bulk_app_no_wsid(self, c):
 
     ############# End tests for run_app_bulk #############
 
-    @mock.patch(
-        "biokbase.narrative.jobs.appmanager.specmanager.clients.get", get_mock_client
-    )
+    @mock.patch(CLIENTS_AM_SM, get_mock_client)
     def test_app_description(self):
         desc = self.am.app_description(self.good_app_id, tag=self.good_tag)
         self.assertIsInstance(desc, HTML)
 
-    @mock.patch(
-        "biokbase.narrative.jobs.appmanager.specmanager.clients.get", get_mock_client
-    )
+    @mock.patch(CLIENTS_AM_SM, get_mock_client)
     def test_app_description_bad_tag(self):
         with self.assertRaises(ValueError):
             self.am.app_description(self.good_app_id, tag=self.bad_tag)
 
-    @mock.patch(
-        "biokbase.narrative.jobs.appmanager.specmanager.clients.get", get_mock_client
-    )
+    @mock.patch(CLIENTS_AM_SM, get_mock_client)
     def test_app_description_bad_name(self):
         with self.assertRaises(ValueError):
             self.am.app_description(self.bad_app_id)
 
-    @mock.patch(
-        "biokbase.narrative.jobs.appmanager.specmanager.clients.get", get_mock_client
-    )
-    @mock.patch("biokbase.narrative.jobs.specmanager.clients.get", get_mock_client)
+    @mock.patch(CLIENTS_AM_SM, get_mock_client)
+    @mock.patch(CLIENTS_SM, get_mock_client)
     def test_validate_params(self):
         inputs = {
             "reads_tuple": [
                 {
-                    "input_reads_label": "reads file 1",
-                    "input_reads_obj": "rhodobacterium.art.q20.int.PE.reads",
+                    "input_reads_label": READS_FILE_1,
+                    "input_reads_obj": READS_OBJ_1,
                     "input_reads_metadata": {"key1": "value1"},
                 },
                 {
-                    "input_reads_label": "reads file 2",
-                    "input_reads_obj": "rhodobacterium.art.q10.PE.reads",
+                    "input_reads_label": READS_FILE_2,
+                    "input_reads_obj": READS_OBJ_2,
                     "input_reads_metadata": {"key2": "value2"},
                 },
             ],
             "output_object": "MyReadsSet",
-            "description": "New Reads Set",
+            "description": NEW_READS_SET,
         }
         app_id = "NarrativeTest/test_create_set"
         tag = "dev"
-        sm = specmanager.SpecManager()
-        spec = sm.get_spec(app_id, tag=tag)
+        spec = self.am.spec_manager.get_spec(app_id, tag=tag)
+        spec_params = self.am.spec_manager.app_params(spec)
```

Comment:
rather than creating a new spec manager instance, use the one created by the app manager